ledger/sandbox-classic/src/main/scala/platform/sandbox/ConfigConverter.scala

```diff
@@ -0,0 +1,106 @@
+  ParticipantIndexerConfig,
+  ParticipantRunMode,
+}
+import ledger.sandbox.{BridgeConfig, BridgeConfigProvider}
+import lf.language.LanguageVersion
+import platform.common.LedgerIdMode
+import platform.sandbox.config.SandboxConfig.EngineMode
+import scalaz.syntax.tag._
+import scala.jdk.DurationConverters._
+
+import java.util.UUID
+
+object ConfigConverter {
+  private val DefaultH2SandboxJdbcUrl = "jdbc:h2:mem:sandbox;db_close_delay=-1"
+
+  def toSandboxOnXConfig(
+      sandboxConfig: SandboxConfig,
+      maybeLedgerId: Option[String],
+  ): Config[BridgeConfig] = {
+    val singleCombinedParticipant = ParticipantConfig(
+      mode = ParticipantRunMode.Combined,
+      participantId = sandboxConfig.participantId,
+      shardName = None,
+      address = sandboxConfig.address,
+      port = sandboxConfig.port,
+      portFile = sandboxConfig.portFile,
+      // When missing, sandbox-classic used an in-memory ledger.
+      // For Sandbox-on-X we don't offer that, so default to H2
+      serverJdbcUrl = sandboxConfig.jdbcUrl.getOrElse(DefaultH2SandboxJdbcUrl),
+      managementServiceTimeout = sandboxConfig.managementServiceTimeout,
+      // TODO SoX-to-sandbox-classic: Wire up all indexer configurations
+      indexerConfig = ParticipantIndexerConfig(
+        allowExistingSchema = false,
+        inputMappingParallelism = sandboxConfig.maxParallelSubmissions,
+      ),
+    )
+
+    val extraBridgeConfig = BridgeConfig(
+      conflictCheckingEnabled = true,
+      implicitPartyAllocation = sandboxConfig.implicitPartyAllocation,
+      authService = sandboxConfig.authService,
+      timeProviderType =
+        sandboxConfig.timeProviderType.getOrElse(SandboxConfig.DefaultTimeProviderType),
+      // TODO SoX-to-sandbox-classic: Dedicated submissionBufferSize CLI param for sanbox-classic
+      submissionBufferSize = sandboxConfig.maxParallelSubmissions,
+      // TODO SoX-to-sandbox-classic: Dedicated submissionBufferSize CLI param for sanbox-classic
+      maxDedupSeconds = BridgeConfigProvider.defaultExtraConfig.maxDedupSeconds,
+    )
+
+    val allowedLanguageVersions = sandboxConfig.engineMode match {
+      case EngineMode.Stable => LanguageVersion.StableVersions
+      case EngineMode.EarlyAccess => LanguageVersion.EarlyAccessVersions
+      case EngineMode.Dev => LanguageVersion.DevVersions
+    }
+
+    Config[BridgeConfig](
+      mode = Mode.Run,
+      ledgerId = sandboxConfig.ledgerIdMode match {
+        case LedgerIdMode.Static(ledgerId) => ledgerId.unwrap
+        case LedgerIdMode.Dynamic => maybeLedgerId.getOrElse(UUID.randomUUID().toString)
+      },
+      commandConfig = sandboxConfig.commandConfig,
+      submissionConfig = sandboxConfig.submissionConfig,
+      tlsConfig = sandboxConfig.tlsConfig,
+      participants = Seq(
+        singleCombinedParticipant
+      ),
+      maxInboundMessageSize = sandboxConfig.maxInboundMessageSize,
+      configurationLoadTimeout = sandboxConfig.configurationLoadTimeout,
```

Comment:
There is a parameter called `delayBeforeSubmittingLedgerConfiguration` which does not have a counterpart in sandbox-on-x. The tests that use SanboxFixture used to set it to ZERO, it is also a value used underneath in SoX. I suggest removing that parameter as it is effectively ignored

Reply:
this one has a non-zero value when used in `sandbox-next`: https://github.com/digital-asset/daml/blob/main/ledger/sandbox/src/test/suite/scala/platform/sandboxnext/services/completion/CompletionServiceWithEmptyLedgerIT.scala#L42